crates/context/interface/src/journaled_state/account.rs

```diff
@@ -137,6 +137,17 @@ impl<'a, ENTRY: JournalEntryTr> JournaledAccount<'a, ENTRY> {
         true
     }
 
+    /// Sets the nonce of the account.
+    ///
+    /// Touches the account in all cases.
+    #[inline]
+    pub fn set_nonce(&mut self, nonce: u64) {
+        self.touch();
+        self.account.info.set_nonce(nonce);
+        self.journal_entries
+            .push(ENTRY::nonce_changed(self.address));
```

Comment:
We need new entry for this, nonce bump by one is only one that happen in EVM.

I am fine to change curenct nonce_changed entry to contain previous nonce value, to facilitate this.